sahi/slicing.py

```diff
@@ -30,7 +30,7 @@
 def get_slice_bboxes(
     image_height: int,
     image_width: int,
-    slice_height: int = 512,
+    slice_height: int = None,
```

Comment:
Well, I think there can be better alternative instead of this. My thought is if we gonna make this `None`, make both `slice_height` and `slice_width` `None`, and introduce a new parameter for auto compute, like the following

```python
def get_slice_bboxes(
  ...,
  slice_height: int = None,
  slice_width: int = None,
  auto_slice_resolution: bool = True,
  ...,
)

if slice_height and slice_width:  # If both are not None
  y_overlap = int(overlap_height_ratio * slice_height)
  x_overlap = int(overlap_width_ratio * slice_width)
elif auto_slice_resolution: 
  x_overlap, y_overlap, slice_width, slice_height = calc_slice_params(height=image_height, width=image_width)
else:
  raise ValueError("Compute type is not auto and no values are provided.")  # Mine is just a placeholder message
```

, or alternatively

```python
def get_slice_bboxes(
  ...,
  slice_height: int = 512,
  slice_width: int = 512,
  auto_slice_resolution: bool = False,
  ...,
)

if auto_slice_resolution:  
  x_overlap, y_overlap, slice_width, slice_height = calc_slice_params(height=image_height, width=image_width)
elif slice_height and slice_width:  # If both are not None
  y_overlap = int(overlap_height_ratio * slice_height)
  x_overlap = int(overlap_width_ratio * slice_width)
else:
  raise ValueError("Compute type is not auto and no values are provided.")  # Mine is just a placeholder message
```
In the second implementation, we preserve default values as `512`, but if `auto_slice_resolution` is `True`, then it disregards whatever `slice_height` and `slice_width` is given.

wdyt ? @mcvarer @fcakyon